packages/frontend/src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,76 @@
+import * as Dialog from '@radix-ui/react-dialog'
+import { ReactNode, useCallback } from 'react'
+import { CloseButton } from 'src/components/Buttons/CloseButton'
+import { hexOpacity, Colors } from 'src/styles/colors'
+import styled from 'styled-components'
+
+export interface ModalProps {
+  isShown: boolean | undefined
+  onRequestClose: () => void
+  title?: string
+  children: ReactNode
+}
+
+export const Modal = ({ isShown, onRequestClose, title, children }: ModalProps) => {
+  const closeModal = useCallback(() => isShown && onRequestClose(), [isShown, onRequestClose])
+
+  return (
+    <Dialog.Root open={isShown}>
+      <Dialog.Portal>
+        <Overlay onClick={closeModal}>
```

Comment:
Clicking anywhere within the modal closes it for me. I'd change it so that only clicking the dimmed background closes the modal.